src/components/BookForm.js

```diff
@@ -64,19 +70,25 @@ class BookForm extends React.Component {
     } else {
       value = e.currentTarget.value;
     }
-    this.props.editBook(name, value);
+    if (name === "isbn") {
+      this.props.checkForCollision(value);
+    }
+    if (value) {
+      book[name] = value;
+    } else {
+      delete book[name];
+    }
+    this.setState({ book });
   };
 
   render() {
-    const { ...book } = this.props.bookFormContent;
-    let classes = "modal";
-    if (this.props.showEditForm) classes += " is-visible";
+    const { book } = this.state;
     let handler = this.handleSubmit;
-    if (this.props.isbnCollision && this.props.editing === false) {
+    if (this.props.isbnCollision && !this.props.selectedBook.isbn) {
```

Comment:
Wow, that performance change you made is also making the code more readable! You were right about what you said about React letting you know when you're fighting against it. :)